kytos/core/auth.py

```diff
@@ -44,6 +56,120 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+@for_all_methods(
+    retries,
+    stop=stop_after_attempt(
+        int(os.environ.get("MONGO_AUTO_RETRY_STOP_AFTER_ATTEMPT", 3))
+    ),
+    wait=wait_random(
+        min=int(os.environ.get("MONGO_AUTO_RETRY_WAIT_RANDOM_MIN", 0.1)),
+        max=int(os.environ.get("MONGO_AUTO_RETRY_WAIT_RANDOM_MAX", 1)),
+    ),
+    before_sleep=before_sleep,
+    retry=retry_if_exception_type((AutoReconnect,)),
+)
+class UserController:
+    """UserController"""
+
+    # pylint: disable=unnecessary-lambda
+    def __init__(self, get_mongo=lambda: Mongo()):
+        self.mongo = get_mongo()
+        self.db_client = self.mongo.client
+        self.db = self.db_client[self.mongo.db_name]
+
+    def bootstrap_indexes(self):
+        """Bootstrap all users related indexes."""
+        index_tuples = [
+            ("users", [("username", pymongo.ASCENDING)], {"unique": True}),
+        ]
+        for collection, keys, kwargs in index_tuples:
+            if self.mongo.bootstrap_index(collection, keys, **kwargs):
+                LOG.info(
+                    f"Created DB index {keys}, collection: {collection}"
+                )
+
+    def create_user(self, user_data: dict) -> InsertOneResult:
+        """Create user to database"""
+        try:
+            utc_now = datetime.datetime.utcnow()
+            result = self.db.users.insert_one(UserDoc(**{
+                **user_data,
+                **{"inserted_at": utc_now},
```

Comment:
@Alopalao, when instantiating `UserDoc`, let's also set a `_id` to avoid storing `id` as `null`, we're currently relaying on `username` and allowing MongoDB to automanage the `_id` (which works great), but if we can set upfront an `str(uuid.uuid4())` that'd be great and would give us possibility to also work with the `_id` in the future, and have `id` also set to the same value, which is a pattern that we also have on other collections.

Reply:
Got it.